src/main/webapp/3Dscript.js

```diff
@@ -176,11 +239,93 @@ function plotOrientation() {
   orientation.rotation.y = THREE.Math.degToRad(poseTransform.rotate);
 }
 
+/**
+* get time in readable units
+* @param {double} time raw time up to 6 decimal places. from GPS time
+* @return {String} readable time.
+ */
+function formatTime(time) {
```

Comment:
Since Jie recommended displaying raw time, I'll just suggest that you may still want to format the time by adding 1000s separators in order to make large quantities easier to read. The rest of this comment is just FYI.

I'm not sure GPS and Unix time can be converted with a fixed offset. There may be leap second or other shenanigans, which are common when dealing with time. In general it is _very_ difficult to get time right, so you'll want to use well-tested library functions whenever possible in favor of doing your own conversions. My team has a saying when investigating a bug: it's always timestamps.